packages/space-header/src/index.ts

```diff
@@ -9,7 +9,12 @@ import { inject } from "./inject";
 async function main(initialSpace: string | Space, options?: Options) {
 	if (window === undefined) return console.error("Please run this script in a browser environment");
 	// Don't run on huggingface.co to avoid duplicate headers
-	if (window.location?.origin === "https://huggingface.co") return;
+	const has_huggingface_ancestor = Object.values(
+		window.location?.ancestorOrigins ?? {
+			0: window.document.referrer,
+		}
+	).some((origin) => origin.includes("huggingface.co"));
```

Comment:
I think the `includes` check is too simplistic, eg `abchuggingface.co` would work.

Is it possible to have a better check, eg with `new URL(x).origin === "https://huggingface.co"` or something?

Reply:
Yes sure
I switched to a .startsWith, because referrer could be https://huggingface.co/spaces/xxxx, while ancestorOrigins is just the domain origin, so should work in both case

Reply:
`startsWith` is also vulnerable - eg `huggingface.co.evil.com`, can we use `new URL(x).origin` and do an equality check?

Reply:
should be good! cc @coyotte508 tell me what do you think